ledger-tool/src/main.rs

```diff
@@ -1560,34 +1560,65 @@ fn main() {
+
+                    let mut packet_count = 0;
+                    /*
+                    let mut events = vec![];
+                    */
+                    use solana_sdk::message::AddressLoader;
+                    use solana_sdk::message::v0::LoadedAddresses;
+                    use solana_sdk::message::v0::MessageAddressTableLookup;
+                    #[derive(Clone)]
+                    struct A;
+                    impl AddressLoader for A {
+                        fn load_addresses(self, _: &[MessageAddressTableLookup]) -> Result<LoadedAddresses, solana_sdk::message::AddressLoaderError> { Ok(LoadedAddresses::default()) }
                     }
-                    let genesis_config = open_genesis_config_by(&ledger_path, arg_matches);
-                    let blockstore = open_blockstore(
-                        &ledger_path,
-                        arg_matches,
-                        get_access_type(&process_options),
-                    );
-                    let (bank_forks, _) = load_and_process_ledger_or_exit(
-                        arg_matches,
-                        &genesis_config,
-                        Arc::new(blockstore),
-                        process_options,
-                        snapshot_archive_path,
-                        incremental_snapshot_archive_path,
-                    );
 
-                    println!(
-                        "{}",
-                        compute_shred_version(
-                            &genesis_config.hash(),
-                            Some(&bank_forks.read().unwrap().working_bank().hard_forks())
-                        )
-                    );
+
+                    loop {
+                        let d = bincode::deserialize_from::<_, TimedTracedEvent>(&mut stream);
+                        let Ok(event) = d else {
+                            info!("deserialize error: {:?}", &d);
+                            break;
+                        };
+                        error!("{:?}", event);
+                        let event_time = event.0;
+                        let event = &event.1;
+                        let datetime: chrono::DateTime<chrono::Utc> = event_time.into();
+                        match &event {
+
+                            TracedEvent::PacketBatch(_label, batch) => {
+                                let sum = batch.0.iter().map(|v| v.len()).sum::<usize>();
+                                packet_count += sum;
+                                let st = solana_core::banking_stage::packet_deserializer::PacketDeserializer::deserialize_and_collect_packets(0, &[batch.clone()], false).deserialized_packets.iter().filter_map(|ip| ip.build_sanitized_transaction(&Arc::new(FeatureSet::all_enabled()), false, A)).collect::<Vec<_>>();
+                                let st_label = format!("({:#?})", st);
+                                for transaction in st {
+                                    solana_cli_output::display::println_transaction(
```

Comment:
this didn't decode instruction data for loaders like this. i wonder there's better way...